### DISCUSSION TITLE
Self-Referencing Field Aliases To Segregate Out Sub-Objects

### DISCUSSION THREAD

**Discussion by parkerhancock** [pydantic/pydantic#8048] [Question]
I'm using (and loving!) pydantic in some data parsing projects. One issue I've run into is that I'm getting very large and "flat" JSON objects, like this:

```json
{
    "player_firstname": "John",
    "player_lastname": "Smith",
    "coach_firstname": "Mark",
    "coach_lastname": "Jenkins",
}
```
And I want to parse it into Pydantic objects that, when dumped, look like this:
```json
{
    "player": {
        "first_name": "John",
        "last_name": "Smith"
    },
    "coach": {
        "first_name": "Mark",
        "last_name": "Jenkins",
    }
}
```
Is there a way to do this using purely pydantic? Something like (and I know this doesn't work):
```python
class Player(BaseModel):
    first_name: str = Field(alias="player_firstname")
    last_name: str = Field(alias="player_lastname")

class Coach(BaseModel):
    first_name: str = Field(alias="coach_firstname")
    last_name: str = Field(alias="coach_lastname")

class Object(BaseModel):
    player: Player = Field(alias=[Self?])
    coach: Coach = Field(alias=[Self?])
```

Is there something I can put for "Self" that works? Or some variant of `AliasPath` or `AliasChoices` (or something like that) that would work as an `AliasSelf`?

For the sake of conversation, assume the real data here has >200 fields that should actually be segregated out into about a dozen sub-objects, and that writing a function to do data reshaping on the input without Pydantic would be complex and hard to maintain.

Thanks!

**Comment by JesseDeLoore** [comment]
Did you ever find a workaround for this ?
